invisible_cities/cities/components.py

```
@@ -179,6 +179,8 @@ def index_tables(file_out):
     -checks if any columns in the tables have been marked to be indexed by writers
     -indexes those columns
     """
+    if not os.path.exists(file_out):
+        return
```

Please move this check to the function `city`. This function shouldn't exit silently if the input is missing. It's in the cities where we want to handle that case.

---

How do we want to handle this case? Do we want an exception to be raised?

---

Here we don't. If the file doens't exist, the function should raise an exception. In `city` we know that there might not be an output file and so we should call `index_tables` only if the file exists.